coalib/core/PersistentHash.py

```diff
@@ -2,18 +2,50 @@
 import pickle
 
 
-def persistent_hash(obj):
+def order(obj):
+    """
+    Recursively preserves the order of any sets or dicts
+    found in obj.
     """
-    Calculates a persistent hash of the given object.
+    if isinstance(obj, set):
+        return sorted(obj)
+    elif isinstance(obj, dict):
```

Comment:
since you return above, you can make this a normal `if` instead. (Btw: I've seen that many linters suggest to start a new `if` in case you return inside the first branch, so I tend to follow that recommendation :smile:)